File: samples/sum_of_odds/sum_of_odds.cs
// sum_of_odds.cs - calculate sum of odd integers

using System;

class sum_of_odds
{
  public static readonly double version = 1.1;
  public static readonly string date = "2021-01-14";

  public static readonly int up_to = 170;

  public static bool is_odd(int x)
  {
    int remainder;

    remainder = x % 2;
    if (remainder == 0) return false;
    else                return true;
  }

  static void Main(string[] args)
  {
    int sum = 0;
    int number_of_odds = 0;
    int last_odd = 0;

    for (int i = 0; i <= up_to; i++)
    {
      if (is_odd(i))
      {
        sum += i;
        number_of_odds += 1;
        last_odd = i;
      }
    }

   if (sum == Math.Pow(number_of_odds, 2))
   {
     if (number_of_odds == 0)
       sum = 0;
     else if (number_of_odds < 7)
       Console.WriteLine(sum);
     else
       Console.WriteLine(
           "1 + 3 + 5 + ... + {0} + {1} = {2}",
           last_odd - 1, last_odd, sum);
   }
   else
     Console.Error.WriteLine("Unexpected error: {0}", sum);
  }
}

File: samples/sum_of_odds/sum_of_odds.go
// sum_of_odds.go - calculate sum of odd integers

package main

import (
  "os"
  "fmt"
  "math"
  "strconv"
)

const __version__ = 1.1;
const __date__ = "2021-01-19";

const up_to = 170

func is_odd(x int) bool {
  var remainder int

  remainder = x % 2
  if remainder == 0 {
    return false
  } else {
    return true
  }
}

func main() {
  var sum int = 0
  var number_of_odds int = 0
  last_odd := 0	// can omit int

  for i := 0; i <= up_to; i++ {
    if is_odd(i) {
      sum += i; number_of_odds++
      last_odd = i
    }
  }

  if float64(sum) == math.Pow(
         float64(number_of_odds), 2.0) {
    if (number_of_odds == 0) {	// do nothing
    } else if number_of_odds < 7 {
      fmt.Println(sum)
    } else {
      fmt.Println("1 + 3 + 5 + ... +",
          last_odd - 1, "+", last_odd, "=", sum)
    }
  } else {
    var message string
    message = "Unexpected error: " +
                  strconv.Itoa(sum) + "\n"
    os.Stderr.WriteString(message)
  }
}
